### PR TITLE
CLI Script changes to fix Corfu Tutorial in README

### PR DESCRIPTION
**Pull request by nvaishampayan517**
## Overview

Description:
Fixed corfu_bootstrap_cluster CLI script to handle layouts with missing Cluster IDs by generating a new UUID. Removed the broken edit functionality from corfu_layouts CLI script. Updated README to show the use of add_node and remove_node as opposed to manually changing the layout.

Why should this be merged: 

Fixes issues blocking another team from deploying CorfuDB.

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [ ] There are no TODOs left in the code
- [ ] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [ ] Change is covered by automated tests
- [ ] Public API has Javadoc

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,9 +17,9 @@ software architecture and example usage.
 ### Table of Contents
```
Comment (Maithem) [inline README.md:17]: you can remove `https://github.com/CorfuDB/CorfuDB/tree/cli_script_bugfixes#install-from-debian-package`

Comment (Maithem) [reply README.md:17]: It's probably okay to just mention that maven 8 is required and omit the java installation steps.

Comment (Maithem) [reply README.md:17]: You can remove these references as well 

```java

<repositories>
    <repository>
        <id>corfu-mvn-repo</id>
        <url>https://raw.github.com/CorfuDB/Corfu-Repos/mvn-repo/</url>
        <snapshots>
            <enabled>true</enabled>
            <updatePolicy>always</updatePolicy>
        </snapshots>
    </repository>
</repositories>

```

Comment (Maithem) [reply README.md:17]: @xcchang Where do we push the runtime artifacts?

This isn't correct, right?

```java

 <dependency>
    <groupId>org.corfudb</groupId>
    <artifactId>runtime</artifactId>
    <version>0.1-SNAPSHOT</version>
    <scope>compile</scope>
</dependency>
```

File: README.md [hunk 2]
```diff
@@ -238,38 +240,27 @@ a single failure.
 
 To learn more about segments, see the [Corfu wiki](https://github.com/CorfuDB/CorfuDB/wiki).
 
-To scale Corfu, we add additional ``stripes''. To add an additional stripe, first 
-start a new ```corfu_server``` on port 9002:
+To add more nodes, begin by 
+starting a new ```corfu_server``` on port 9002:
 ```
 ./CorfuDB/bin/corfu_server -m 9002
 ```
```
Comment (Maithem) [inline README.md:247]: I think it would be cleaner to just use `/bin/corfu_remove_node` and assume that the user is in the parent directory.

**Review by Maithem — APPROVED** [review]
Looks great! Just added some minor comments.

File: README.md (hunk 1 above)
Comment (Maithem) [reply README.md:17]: Also, the example in https://github.com/CorfuDB/CorfuDB/tree/cli_script_bugfixes#now-i-want-to-write-a-program-that-uses-corfu doesn't make a lot of sense. It should print the value of the map.get before and after to demonstrate the db access.

File: README.md (hunk 2 above)
Comment (nvaishampayan517) [reply README.md:220]: this is done for consistency with the rest of the command examples, which are all in the parent dir of CorfuDB

File: README.md (hunk 1 above)
Comment (nvaishampayan517) [reply README.md:17]: > Also, the example in https://github.com/CorfuDB/CorfuDB/tree/cli_script_bugfixes#now-i-want-to-write-a-program-that-uses-corfu doesn't make a lot of sense. It should print the value of the map.get before and after to demonstrate the db access.

I'm not sure what you mean by this change. The current code shows the db access across clients by printing out the value of the variable currently stored.

File: README.md (hunk 2 above)
Comment (Maithem) [reply README.md:220]: I meant change all of them.

File: README.md (hunk 1 above)
Comment (Maithem) [reply README.md:17]: I see. Just make sure the dependency is fixed. We don't push to `//raw.github.com/CorfuDB/Corfu-Repos/mvn-repo`.
@xcchang made this change, I think.

Comment (xcchang) [reply README.md:17]: > @xcchang Where do we push the runtime artifacts?
> 
> This isn't correct, right?
> 
> ```java
>  <dependency>
>     <groupId>org.corfudb</groupId>
>     <artifactId>runtime</artifactId>
>     <version>0.1-SNAPSHOT</version>
>     <scope>compile</scope>
> </dependency>
> ```

We are pushing to GitHub Packages now: https://github.com/orgs/CorfuDB/packages?repo_name=CorfuDB. No the version string should be `0.3.1-SNAPSHOT`.

File: README.md [hunk 3]
```diff
@@ -238,38 +240,27 @@ a single failure.
...
+You will see that the cluster contains the new node.
 
-This adds the logunit at localhost:9002 as an additional stripe in the system. That is, writes to even addresses will now go to
-localhost:9000 and localhost:9001, while writes to odd addresses will go to localhost:9002.
+To remove the added node from the cluster:
+```
+./CorfuDB/bin/corfu_remove_node -c localhost:9000,localhost:9001,localhost:9002 -n localhost:9002
+```
 
 ## Now I want to write a program that uses Corfu!
```
Comment (xcchang) [inline README.md:239]: I known it's not from your PR. I think there are problems here. 

- The exposed version string for every module in `CorfuDB` repo is `0.3.1-SNAPSHOT`. Currently we are just using SNAPSHOT with timestamp to update our artifacts, which is not visible outside.
- The "You will also want to add the Corfu Maven repository" part might not be valid now. As GitHub Packages have a known [issue](https://github.community/t/download-from-github-package-registry-without-authentication/14407) , authentication are required for even public artifacts. So might be good to give the links for authenticating like [this for Maven](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry#authenticating-to-github-packages) and [this for Gradle](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-gradle-registry).
- For importing dependencies part, I think only adding `runtime` module is not enough, normally we need everything [here](https://github.com/orgs/CorfuDB/packages?repo_name=CorfuDB). How about give the link here, and use `runtime` as an example? The Packages page provides code snippets for importing them actually (after authentication).

**Review by xcchang — APPROVED** [review]
LGTM!

File: corfu_scripts/corfu_layouts.clj [hunk 4]
```diff
@@ -30,68 +29,8 @@ Options:
 (connect-runtime)
 (def layout-view (get-layout-view))
 
-(defn install-layout
```
[the remark this answers is not in the document]
Comment (nvaishampayan517) [reply corfu_scripts/corfu_layouts.clj:33]: no the functionality is being removed. The philosophy is that the CLI should only expose creating the cluster, and adding and removing nodes.

File: corfu_scripts/corfu_bootstrap_cluster.clj [hunk 5]
```diff
@@ -5,6 +5,7 @@
 (require 'clojure.pprint)
 (require 'clojure.java.shell)
 (import org.corfudb.runtime.view.Layout)
+(import java.util.UUID)
 (def usage "corfu_bootstrap_cluster, setup the Corfu cluster where all nodes are NOT bootstrapped.
```
[the remark this answers is not in the document]
Comment (nvaishampayan517) [reply corfu_scripts/corfu_bootstrap_cluster.clj:9]: I thought it meant set up the Corfu cluster, from nodes that are not already bootstrapped

[the remark this answers is not in the document]
Comment (nvaishampayan517) [reply corfu_scripts/corfu_bootstrap_cluster.clj:9]: I suppose a clearer phrasing would be "setup the Corfu cluster from nodes that have NOT been bootstrapped"?

**Review by pankti-m — APPROVED** [review]

**Pull request merged**